catalog/files/impl/src/main/java/org/projectnessie/catalog/files/adls/AdlsFileSystemOptions.java

```diff
@@ -64,4 +67,10 @@ enum AdlsRetryStrategy {
     EXPONENTIAL_BACKOFF,
     FIXED_DELAY,
   }
+
+  enum AzureAuthType {
+    STORAGE_SHARED_KEY,
```

Comment:
Can you add a `NONE` option as well and let `AdlsClientSupplier.fileSystemClient()` default to `NONE`?

Although it's probably rare that file-systems are publicly accessible, and even unlikely that those are writeable, but we can't predict which auth method users want.

Reply:
Should I throw `IllegalArgumentException("Unsupported auth type " + authType);` if it is `NONE` as `clientBuilder.credential` will throw `NPE` if credential is null